runtime/kusama/src/weights/runtime_parachains_paras.rs

```diff
@@ -146,28 +145,32 @@ impl<T: frame_system::Config> runtime_parachains::paras::WeightInfo for WeightIn
 		// Proof Size summary in bytes:
 		//  Measured:  `4251`
 		//  Estimated: `7716`
-		// Minimum execution time: 20_280_000 picoseconds.
-		Weight::from_parts(20_641_000, 0)
+		// Minimum execution time: 19_150_000 picoseconds.
+		Weight::from_parts(19_571_000, 0)
 			.saturating_add(Weight::from_parts(0, 7716))
 			.saturating_add(T::DbWeight::get().reads(2))
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
-	/// Storage: Paras PvfActiveVoteMap (r:1 w:0)
+	/// Storage: Paras PvfActiveVoteMap (r:1 w:1)
 	/// Proof Skipped: Paras PvfActiveVoteMap (max_values: None, max_size: None, mode: Measured)
-	/// Storage: Paras CodeByHash (r:1 w:1)
-	/// Proof Skipped: Paras CodeByHash (max_values: None, max_size: None, mode: Measured)
+	/// Storage: Paras PvfActiveVoteList (r:1 w:1)
+	/// Proof Skipped: Paras PvfActiveVoteList (max_values: Some(1), max_size: None, mode: Measured)
+	/// Storage: ParasShared CurrentSessionIndex (r:1 w:0)
+	/// Proof Skipped: ParasShared CurrentSessionIndex (max_values: Some(1), max_size: None, mode: Measured)
+	/// Storage: Paras ActionsQueue (r:1 w:1)
+	/// Proof Skipped: Paras ActionsQueue (max_values: None, max_size: None, mode: Measured)
 	/// The range of component `c` is `[1, 3145728]`.
 	fn add_trusted_validation_code(c: u32, ) -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `28`
-		//  Estimated: `3493`
-		// Minimum execution time: 8_193_000 picoseconds.
-		Weight::from_parts(8_317_000, 0)
-			.saturating_add(Weight::from_parts(0, 3493))
+		//  Measured:  `622`
+		//  Estimated: `4087`
+		// Minimum execution time: 80_323_000 picoseconds.
+		Weight::from_parts(57_788_688, 0)
```

Comment:
Where is this weight explosion coming from?

Reply:
Potentially `add_trusted_validation_code` concludes prechecking, which causes a few storage accesses. I intentionally extended the benchmark to account for it. See https://github.com/paritytech/polkadot/pull/6347 for more details.

In fact, this dispatchable is only really needed for tests: https://github.com/paritytech/polkadot/pull/7138#discussion_r1177993609 but I decided not to drop it in the course of this pr. Later we can `cfg` it to be only present in test Runtime or when `benchmarks` feature is enabled.